packages/base/src/util/encodeXML.js

```diff
@@ -0,0 +1,35 @@
+const fnToHex = (iChar, iLength) => {
+	let sHex = iChar.toString(16);
+	if (iLength) {
+		sHex = sHex.padStart(iLength, "0");
+	}
+	return sHex;
+};
+
+const rHtml = /[\x00-\x2b\x2f\x3a-\x40\x5b-\x5e\x60\x7b-\xff\u2028\u2029]/g, // eslint-disable-line
+	rHtmlReplace = /[\x00-\x08\x0b\x0c\x0e-\x1f\x7f-\x9f]/, // eslint-disable-line
+	mHtmlLookup = {
+		"<": "&lt;",
+		">": "&gt;",
+		"&": "&amp;",
+		"\"": "&quot;",
```

Comment:
**[suggestion (non-blocking)](https://conventionalcomments.org/#labels):** Per https://en.wikipedia.org/wiki/Character_encodings_in_HTML#XML_character_references for XML, let's include a `'` to `&apos;` mapping too...

```suggestion
	mHtmlLookup = {
		"<": "&lt;",
		">": "&gt;",
		"&": "&amp;",
		"\"": "&quot;",
		"'": "&apos;"
```

Reply:
@stoehr ,
Please take a look at https://stackoverflow.com/questions/2083754/why-shouldnt-apos-be-used-to-escape-single-quotes

While technically all modern browsers do support this HTML entity with HTML5, can you justify the reason why we should use an HTML entity that is not officially supported by HTML4 spec?

Is there a real use case for it?

Reply:
So we agreed that `encodeXML` should just match precisely the same encodeXML.js from openUI5, at least the logic will be the same (but now changed to es6 module).

Reply:
> While technically all modern browsers do support this HTML entity with HTML5, can you justify the reason why we should use an HTML entity that is not officially supported by HTML4 spec?
>
> Is there a real use case for it?

My reasoning is centered around having one XML encoding code in the "ui5-webcomponents" core that can be re-used for HTML encoding too, and then in the future, will be available for XML-compliant encoding.

> So we agreed that encodeXML should just match precisely the same encodeXML.js from openUI5, at least the logic will be the same (but now changed to es6 module).

Yeah, my vote is to have encodeXML be in "ui5-webcomponents" like it is in the Open UI5 Framework. Then both projects can have the same impl for encoding of XML and HTML, which should make it easier to apply security bug fixes to both projects in the future, if and when needed.